### PR TITLE
Bugfix : AN-6825 - Logging out one of multiple accounts does not land on conversations screen.

### PR DESCRIPTION
**Pull request by mejdoo**
## What's new in this PR?

https://wearezeta.atlassian.net/browse/AN-6825

### Issues

Logging out one of multiple accounts does not land on conversations screen.

### Causes

The current code is calling **twice** the `back` method of `BackStackNavigator` component in oder to achieve the logout when we have multiple accounts. This may lead to a weird behaviours of the client in some testing scenario.

Example of failure reports : https://192.168.120.44:8081/view/Android/job/android_pipeline_bug_client/166/cucumber-html-reports/report-feature_3_3894774586.html

### Solutions

Kill `PreferencesActivity` when we need to logout from any active account, and if there is no more active accounts we go back to `AppEntryActivity`

### Testing

**First scenario:** 
- Login with an account
- Logout
 
Expected result: 
I see the welcome screen

**Second scenario :** 
- Login with the 1st account
- Login with the 2nd account
- Logout from the 2nd account
 
Expected result: 
I see the conversations screen (1st account)

**Third scenario :** 
- Login with the 1st account
- Login with the 2nd account
- Logout from the 1st account
 
Expected result: 
I see the conversations screen (2nd account)
#### APK
[Download build #2228](http://10.10.124.11:8080/job/Pull%20Request%20Builder/2228/artifact/build/artifact/wire-dev-PR2888-2228.apk)

### CODE REVIEW

File: app/src/main/scala/com/waz/zclient/preferences/pages/AccountView.scala [hunk 1]
```diff
@@ -332,10 +332,9 @@ class AccountViewController(view: AccountView)(implicit inj: Injector, ec: Event
             .flatMap(_ => accounts.accountsWithManagers.head.map(_.isEmpty)).map {
             case true =>
               context.startActivity(AppEntryActivity.newIntent(context))
-              Option(context.asInstanceOf[Activity]).foreach(_.finish())
+              finishPreferencesActivity()
             case false =>
-              navigator.back()
-              navigator.back()
+              finishPreferencesActivity()
```
Comment (BradleyWire) [inline app/src/main/scala/com/waz/zclient/preferences/pages/AccountView.scala:337]: You can remove the true/false case and change this to `case _ => finishPreferencesActivity()` only if i'm not mistaken

Comment (mejdoo) [reply app/src/main/scala/com/waz/zclient/preferences/pages/AccountView.scala:337]: what about  `context.startActivity(AppEntryActivity.newIntent(context))` I need it for the true case

Comment (BradleyWire) [reply app/src/main/scala/com/waz/zclient/preferences/pages/AccountView.scala:337]: Oh that's my bad! Didn't spot that.

Comment (mejdoo) [reply app/src/main/scala/com/waz/zclient/preferences/pages/AccountView.scala:337]: I couldn't find a better way to do it in Scala. Suggestions are welcome :)

Comment (BradleyWire) [reply app/src/main/scala/com/waz/zclient/preferences/pages/AccountView.scala:337]: No, looks completely fine otherwise as far as my Scala knowledge goes

**Pull request merged**